client/src/pages/GrantAccessForm.js

```diff
@@ -0,0 +1,197 @@
+import { useState } from "react";
+import "./GrantAccessForm.css";
+import { post } from "../TenderClient";
+import Logo from "../assets/images/CTY-logo-rectangle.png";
+
+const GrantAccessForm = () => {
+	const [role, setRole] = useState("bidder");
+	const [buyerDetails, setBuyerDetails] = useState({
+		userType: "buyer",
+		company: "",
+		description: "",
+		address: "",
+		email: "",
+	});
+
+	const [bidderDetails, setBidderDetails] = useState({
+		userType: "bidder",
+		firstName: "",
+		lastName: "",
+		email: "",
+	});
+
+	const [resgisterStatus, setRegisterStatus] = useState("");
+
+	async function postBuyerDetails(buyerData) {
+		try {
+			const data = await post("/mock-api", buyerData);
```

Comment:
I think there is already an endpoint for sign up. Use that endpoint and test that it works.